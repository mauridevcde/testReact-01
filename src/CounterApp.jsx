import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  const handledAdd = () => {
    setCount(count + 1);
  };
  const handledRestar = () => {
    setCount(count - 1);
  };
  const handledReset = () => {
    setCount(value);
  };

  return (
    <div>
      <h1>CounterApp</h1>
      {<h2>{count}</h2>}
      <button onClick={handledAdd}>+1</button>
      <button onClick={handledRestar}>-1</button>
      <button onClick={handledReset}>Reset</button>
    </div>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
