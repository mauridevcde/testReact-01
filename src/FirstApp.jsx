import PropTypes from "prop-types"

export const FirstApp = ({title,subtitle, name}) => {
    return (
        <div>
            <h1 data-testid="test-title">{title}</h1>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
        </div>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    title: "esta es una props default",
    name: "mauricio es default"
};