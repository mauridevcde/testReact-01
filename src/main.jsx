import React from 'react'
import ReactDOM from 'react-dom/client'
import {CounterApp} from './CounterApp'
import {FirstApp} from './FirstApp'
import './app.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
  {/* <FirstApp  title="Hola soy mauricio" />*/}
  { <CounterApp  value={20} />}
  </React.StrictMode>
) 
