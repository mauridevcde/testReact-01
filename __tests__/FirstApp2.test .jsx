import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en <First app/>", () => {
  //   test("Debe hacer match con el snapshot", () => {
  //     const title = "Hola mundo";

  //     const { container } = render(<FirstApp title={title} />);
  //     //console.log(container);
  //     expect(container).toMatchSnapshot(); //snapshot es una imagen de un componente, comprueba si el componente es igual a la imagen
  //   });

  //   test("Debe mostrar el titulo en un H1", () => {
  //     const title = "Hola soy Mauricio";

  //     const { container, getByText } = render(<FirstApp title={title} />);

  //     //getByText es una funcion que busca un elemento por texto
  //     //toBetruthy es una funcion que comprueba si el elemento existe

  //     expect(getByText(title)).toBeTruthy();

  //     const h1 = container.querySelector("h1");

  //     //toContain es una funcion que comprueba si el elemento contiene el texto sin importar si tiene espacios

  //     expect(h1.innerHTML).toContain(title);
  //   });

  test("Debe mostrar el titulo en un H1", () => {
    const title = "Hola soy Mauricio";

    const { getByText, getByTestId } = render(<FirstApp title={title} />);
    //obs
    //toBe se asegura que sea el mismo dato el mismo en todo sentido
    //toBeTruthy es una funcion que comprueba si el elemento existe
    //toContain tiene que contener el texto sin importar si tiene espacios
    expect(getByTestId("test-title").innerHTML).toContain(title);
  });

  test("Debe mostrar el subtitulo por props", () => {
    const title = "Hola soy goku";
    const subtitle = "Hola soy un subtitulo";

    const { getAllByText } = render(<FirstApp title={title} subtitle={subtitle} />);
    //obs
    //toBe se asegura que sea el mismo dato el mismo en todo sentido
    //toBeTruthy es una funcion que comprueba si el elemento existe
    //toContain tiene que contener el texto sin importar si tiene espacios
    //getAllByText en el caso que queramos buscar todos los elementos que contengan el texto, devuelve un array con todos los elementos que contienen el texto

    expect ( screen.getAllByText(subtitle).length ).toBe(2); //lo filtramos por el length para que nos devuelva el numero de elementos que contienen el texto
    //si tenemos mas de un elemento
  });
});
