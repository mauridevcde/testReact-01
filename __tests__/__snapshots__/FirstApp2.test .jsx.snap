// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <First app/> debe hacer match con el snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="test-title"
    >
      Hola, Soy Goku
    </h1>
    <p />
    <p />
  </div>
</div>
`;
